src/main/scala/cromwell/engine/backend/local/SharedFileSystem.scala

```diff
@@ -97,28 +108,85 @@ trait LocalFileSystemOperations {
+  private def stageWorkflowInputs(descriptor: WorkflowDescriptor): Try[HostInputs] = {
     val hostInputsPath = Paths.get(LocalBackend.hostExecutionPath(descriptor).toFile.getAbsolutePath, "workflow-inputs")
-    descriptor.actualInputs map {case(name, value) => name -> stageWdlValue(value, hostInputsPath)}
+    val attemptedStagedInputs = descriptor.actualInputs map { case(name, value) =>
+      val call = descriptor.namespace.resolve(name.split("\\.").dropRight(1).mkString(".")) match {
+        case Some(c:Call) => Some(c)
+        case _ => None
+      }
+      name -> stageWdlValue(call, value, hostInputsPath)
+    }
+    attemptedStagedInputs collect { case (k:String, f: Failure[_]) => k -> f } match {
+      case m: Map[FullyQualifiedName, Failure[_]] if m.nonEmpty =>
+        val errors = m.values map { _.exception.getMessage } mkString "\n"
+        Failure(new UnsupportedOperationException(s"Failures during localization:\n\n$errors"))
+      case _ =>
+        Success(attemptedStagedInputs map { case (k, v) => k -> v.get})
+    }
   }
 
-  private def stageInputArray(array: WdlArray, hostInputsPath: Path): WdlArray = array.map(stageWdlValue(_, hostInputsPath))
+  private def stageInputArray(call: Option[Call], array: WdlArray, hostInputsPath: Path): Try[WdlArray] = {
+    val attemptedStagedArray = array.value.map(stageWdlValue(call, _, hostInputsPath))
+    attemptedStagedArray collect { case f: Failure[_] => f } match {
+      case s:Seq[Failure[_]] if s.nonEmpty =>
+        val errors = s map { _.exception.getMessage } mkString "\n"
+        Failure(new UnsupportedOperationException(s"Failures during localization of array $array:\n\n$errors"))
+      case _ => Success(WdlArray(array.wdlType, attemptedStagedArray.map(_.get)))
+    }
+  }
 
-  private def stageWdlValue(value: WdlValue, hostInputsPath: Path): WdlValue = value match {
-    case w: WdlFile => stageWdlFile(w, hostInputsPath)
-    case a: WdlArray => stageInputArray(a, hostInputsPath)
-    case x => x
+  private def stageWdlValue(call: Option[Call], value: WdlValue, hostInputsPath: Path): Try[WdlValue] = value match {
+    case w: WdlFile => stageWdlFile(call, w, hostInputsPath)
+    case a: WdlArray => stageInputArray(call, a, hostInputsPath)
+    case x => Success(x)
   }
 
-  private def stageWdlFile(wdlFile: WdlFile, hostInputsPath: Path): WdlFile = {
+  private def stageWdlFile(call: Option[Call], wdlFile: WdlFile, hostInputsPath: Path): Try[WdlFile] = {
     val originalPath = Paths.get(wdlFile.value)
     val executionPath = hostInputsPath.resolve(originalPath.getFileName.toString)
+
+    def invalidLocalization(s: String)(call: Option[Call], originalPath: Path, executionPath: Path): Try[Unit] =
+      Failure(new UnsupportedOperationException(s"Unrecognized localization strategy: $s"))
+
+    val localizers = LocalizationStrategies map {
+      case "hard-link" => localizePathViaHardLink _
```

Comment:
My objection to these sorts of things isn't abstraction and thus the "it's only used in one place" argument doesn't really apply. 

My objection is that if data is coming from the outside world it should be verified immediately and then to use types other than `String` throughout the rest of the system - i.e. abstraction for the sake of strong code sanity, not abstraction for the sake of being modular

If you really want to leave it as-is that's fine, but I'll probably change it the next time I happen to come across it.